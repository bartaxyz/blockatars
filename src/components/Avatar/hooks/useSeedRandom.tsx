import { useMemo } from "react";
import { initSeedValues } from "../utils/initSeedValues";

export const useSeedRandom = (seed: string) => {
  return useMemo(() => {
    const { seedValues } = initSeedValues(seed);

    const shift = () => {
      const t = seedValues[0] ^ (seedValues[0] << 11);

      seedValues[0] = seedValues[1];
      seedValues[1] = seedValues[2];
      seedValues[2] = seedValues[3];
      seedValues[3] = seedValues[3] ^ (seedValues[3] >> 19) ^ t ^ (t >> 8);

      return (seedValues[3] >>> 0) / ((1 << 31) >>> 0);
    };

    return { seedValues, shift };
  }, [seed]);
};
