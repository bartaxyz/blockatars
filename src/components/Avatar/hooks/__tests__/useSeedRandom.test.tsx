// write jest test for useSeedColor hook

import { renderHook } from "@testing-library/react";
import { useSeedRandom } from "../useSeedRandom";

describe("useSeedColor", () => {
  test("returns a color", () => {
    const returnValue = renderHook(() => useSeedRandom("0x1234567890"));

    expect(returnValue.result.current).toMatchSnapshot();
  });
});
