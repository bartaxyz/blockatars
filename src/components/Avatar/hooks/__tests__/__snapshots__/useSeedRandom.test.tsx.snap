// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSeedColor returns a color 1`] = `
{
  "seedValues": [
    47764,
    116988,
    48789,
    49772,
  ],
  "shift": [Function],
}
`;
